census/vanilla-tf/trainer/task.py

```diff
@@ -0,0 +1,338 @@
+
+  while step < max_steps:
+    session.run(
+        train_step,
+        feed_dict={
+            inputs: session.run(train_inp),
+            labels: session.run(train_label)
+            }
+    )
+
+    step = tf.train.global_step(session, global_step)
+
+    if step % 10 == 0:
+      accuracy = evaluate_accuracy(session, eval_step, inputs, labels,
+                                   eval_inp, eval_label)
+      print('[{}/{}]: Step number {} of {} done, Accuracy {:.2f}%'.format(
+          job_name, job_id, step, max_steps, accuracy))
+
+  return train_step
+
+
+#
+# Single and Distributed training functions.
+#
+def training_single(job_dir, max_steps, train_eval_tensor):
+  """Perform single node training."""
+
+  inputs, labels = get_placeholders()
+
+  graph = make_graph(inputs, labels)
+  init = tf.global_variables_initializer()
+
+  session = tf.train.MonitoredTrainingSession(checkpoint_dir=job_dir,
+                                              save_checkpoint_secs=20,
+                                              save_summaries_steps=50)
+  session.run(init)
+
+  training_steps(session, graph, inputs, labels, max_steps, train_eval_tensor)
+
+def training_distributed(job_dir, max_steps, train_eval_tensor):
+  """Perform distributed training."""
+
+  # Parse the TF_CONFIG to create cluster spec and job name
+  # We are doing this manually here to demonstrate
+  # See RunConfig abstraction here https://goo.gl/f4BQMo
+  cluster_spec, job_name, task_index = parse_tf_config()
+
+  # Create and start a server
+  server = tf.train.Server(cluster_spec,
+                           job_name=job_name,
+                           task_index=task_index)
+
+  is_chief = (job_name == 'master')
+
+  if job_name == 'ps':
+    server.join()
+  elif job_name in ['master', 'worker']:
+    with tf.device(tf.train.replica_device_setter()):
+      inputs, labels = get_placeholders()
+      graph = make_graph(inputs, labels)
+
+    init = tf.global_variables_initializer()
+
+    with tf.train.MonitoredTrainingSession(master=server.target,
+                                           is_chief=is_chief,
+                                           checkpoint_dir=job_dir,
+                                           save_checkpoint_secs=20,
+                                           save_summaries_steps=50) as session:
+      session.run(init)
+      training_steps(session, graph, inputs, labels, max_steps,
+                     train_eval_tensor, job_name=job_name, job_id=task_index)
+
+
+def evaluate_accuracy(session, eval_step, inputs, labels,
+                      eval_input, eval_label):
+  """Perform the evaluation step to calculate accuracy."""
+  return 100 * session.run(
+      eval_step,
+      feed_dict={
+          inputs: session.run(eval_input),
+          labels: session.run(eval_label)
+      }
+  )
+
+
+def parse_tf_config():
+  """Parse TF_CONFIG to cluster_spec, job_name and task_index."""
+
+  tf_config = os.environ.get('TF_CONFIG')
```

Comment:
Use a default here -- '{}' + additional defaulting/logic to end up with task_name = 'master' and task_index = 0, in the single non-distributed case. This helps unify the single and distributed modes into common path.